src/olympia/devhub/forms.py

```diff
@@ -178,7 +178,7 @@ def save(self, *args, **kwargs):
             fetch_existing_translations_from_addon(
                 obj, self.fields_to_trigger_content_review)
         )
-        if existing_data != new_data:
+        if obj.has_listed_versions() and existing_data != new_data:
```

Comment:
lets move this up to line 173 so we're not even bothering fetching translations for unlisted only addons.